plugins/postgres/dbt/adapters/postgres/impl.py

```diff
@@ -88,27 +119,50 @@ def check_schema_exists(self, schema, model_name=None):
 
     @classmethod
     def get_columns_in_relation_sql(cls, relation):
+        db_filter = '1=1'
+        if relation.database:
+            db_filter = "table_catalog ilike '{}'".format(relation.database)
+
         schema_filter = '1=1'
         if relation.schema:
             schema_filter = "table_schema = '{}'".format(relation.schema)
 
-        db_prefix = ''
-        if relation.database:
-            db_prefix = '{}.'.format(relation.database)
-
         sql = """
         select
             column_name,
             data_type,
             character_maximum_length,
             numeric_precision || ',' || numeric_scale as numeric_size
 
-        from {db_prefix}information_schema.columns
+        from information_schema.columns
```

Comment:
did you find this db_prefix to be unnecessary? I think that information_schema is database specific in postgres

Reply:
It is invalid on postgres!